Updates to create a user config

dandancheeseandham opened the pull request:
night!

dandancheeseandham committed:
pdates to create a user config

dandancheeseandham committed:
config-user setup

dandancheeseandham committed:
Update config-user.g

dandancheeseandham commented:
I think this is looking decent now.

I'm in the middle of a print, but I will try to test this out this evening if possible (now+6 hours)

dandancheeseandham committed:
minor updates

dandancheeseandham committed:
minor correction to mesh probe and update note for mesh probe

dandancheeseandham committed:
Create README.md

natewalck commented on duet/sys/config-user examples/sensorless/homex.g:
```diff
@@ -6,7 +6,7 @@ M201 X1000 Y1000                ; reduce acceleration on X/Y to stop false trigg
 M915 P0:1 S3 R0 F0              ; both motors because corexy; Sensitivity 4, don’t take action, don’t filter
 
 G91                   	 	; use relative positioning
-G1 S1 X-330 F4000    		; move to home position
```
Regression here, should be 330?

dandancheeseandham replied:
OK, I don't know what regression means in this context :)

This value just needs to be more than X.
Just as a sanity check, I used the RRF configurator and it adds 5 to whatever your defined maximum X or Y is.
Since 310 is possible, we could be 315, but it doesn't hurt to have more does it?

natewalck commented on duet/sys/config.g:
```diff
@@ -72,26 +70,19 @@ M106 P1 S0
-;G10 P1 S0 R0 X0 Y17                    ; Set tool 1 operating and standby temperatures
-
-; Z probe and compensation definition
+M563 P0 S"config-user not set" D0 H1    ; Define tool 0
```
Its pretty safe regardless of build to set the tool, isn't it? It always seemed pretty generic to me.

dandancheeseandham replied:
Ah right. That slipped through :) 
I'd done that before you'd done your bits (which were cool btw :+1:  )

Could make it "M563 tool config not set" :) or just remove, I'm easy.

dandancheeseandham replied:
defined as "Tool 0" otherwise left in as default with -273C as default temps.
I hope thats 0K :rofl:

natewalck commented on duet/sys/config.g:
```diff
@@ -72,26 +70,19 @@ M106 P1 S0
-; *** Conservative settings enabled during build and testing. When you are ready to remove this ***
-; *** simple protection, remove or comment out this line                                        ***
-M98 P"conservative.g"
+M98 P"config-user.g"                    ; Load custom user config
```
Does order here matter? If someone sets a PID tuning here, wouldn't the M501 below override it? Not sure on the best order.

dandancheeseandham replied:
Actually conservative.g should be gone - it's all performed by config.g now.
So good catch regardless

dandancheeseandham replied:
erm, it's not there
config.g only goes to line 88 - are you looking at something else here?

dandancheeseandham replied:
Yes, there isn't both. conservative.g has gone.
So the situation is
config.g -> "load all common stock values, and many with conservative settings"
config-user.g -> "custom values, including speeds,accel,jerk etc to override conservative settings"
I hoping that makes sense to people :)

dandancheeseandham replied:
There is a very rough README at
https://github.com/railcore/configs/blob/config-user/duet/sys/config-user%20examples/README.md
maybe it should be in /sys instead

(yes it will be fleshed out)

dandancheeseandham committed:
define tool 0

dandancheeseandham committed:
comment out heater temp

dandancheeseandham committed:
remove description

dandancheeseandham committed:
additional warning and clarification

dandancheeseandham committed:
removed S description from config.g

dandancheeseandham committed:
reinstated E3D fold settings

after natewalck had a moment of not being precise :rofl:

dandancheeseandham committed:
minor comment spacing

dandancheeseandham committed:
-330 for homing

dandancheeseandham committed:
-330 for homing

dandancheeseandham committed:
Standby and initial Temp for bed as "off"

M140 S-273 R-273 (-273 = "off")

natewalck reviewed (commented):
Looks good to me. Let’s get it tested and merge it in :D

dandancheeseandham committed:
Update README.md

dandancheeseandham committed:
Update README.md

dandancheeseandham committed:
Update README.md

dandancheeseandham committed:
Update config-user.kit

dandancheeseandham committed:
Update config-user.kit

dandancheeseandham committed:
Update config-user.kit

dandancheeseandham committed:
Update config-user.kit

dandancheeseandham committed:
Update config-user.kit

dandancheeseandham committed:
fast track some macros for release

chastise cheeseandham if this is a problem.

dandancheeseandham committed:
reorganise macros 1

dandancheeseandham committed:
reorganise macros

create commissioning folder and reorganise macros.

dandancheeseandham committed:
update names for readability

There is no need for CamelCase in the macros directory, and just reduces readability in DWC

dandancheeseandham committed:
change names due to Windows case issues

dandancheeseandham committed:
Windows upper/lower case issue correction.

dandancheeseandham committed:
README and ordering of macros in the root

added centre and home macros

dandancheeseandham committed:
Update README.md

dandancheeseandham committed:
Update Level Then Probe.g

dandancheeseandham committed:
renumbering macros

dandancheeseandham committed:
Added Kraegars latest Dropbox config for completion

dandancheeseandham committed:
rename Kraegars config

dandancheeseandham committed:
move to subdir for consistency

natewalck reviewed (approved):
Let’s doooooitttttt

dandancheeseandham committed:
Update README.md

dandancheeseandham committed:
Update README.md

dandancheeseandham committed:
snake case

dandancheeseandham committed:
Update README.md

dandancheeseandham committed:
comment spacing

dandancheeseandham commented:
I'm primed and ready :)
